app/javascript/alchemy_admin/dialog.js

```diff
@@ -0,0 +1,119 @@
+import { createHtmlElement } from "alchemy_admin/utils/dom_helpers"
+
+export class Dialog {
+  #dialog
+  #onReject
+  #onResolve
+
+  /**
+   * @param {string} url
+   * @param {object} options
+   */
+  constructor(url, options = {}) {
+    this.url = url
+    this.options = { title: "", size: "300x400", padding: true, ...options }
+  }
+
+  /**
+   * load the content of given url and than open the dialog
+   * @returns {Promise<unknown>}
+   */
+  open() {
+    this.#loadContent().then((content) => {
+      // create the dialog markup and show the dialog
+      this.#build(content)
+      this.#subscribeFormSubmit()
+      this.#dialog.show()
+
+      // bind the current class instance to the DOM - element
+      // this should be an intermediate solution
+      // the main goal, is to close the dialog with the turbo:submit-end - event
+      this.#dialog.connectedDialogInstance = this
+
+      // the dialog is closing with the overlay, esc - key, or close - button
+      this.#dialog.addEventListener("sl-request-close", () => {
+        this.#dialog.remove()
+        this.#onReject()
+      })
+    })
+
+    return new Promise((resolve, reject) => {
+      this.#onResolve = resolve
+      this.#onReject = reject
+    })
+  }
+
+  /**
+   * hide and remove dialog
+   * the open - promise will be resolved
+   */
+  close() {
```

Comment:
Why are we resolving the promise on close here while we call reject above? This does not make any sense to me?

Reply:
Maybe we should rename this method into `onSubmitSuccess()`